spec/grover/processor_spec.rb

```diff
@@ -490,6 +490,25 @@
         it { expect(pdf_text_content).to eq "#{date} Hey there http://www.example.net/foo/bar 1/1" }
       end
 
+      context 'when request failure option is specified' do
+        let(:url_or_html) do
```

Comment:
per above, this likely should be two separate contexts under the "request failure option specified" context (bad host VS bad path) 

I'd also suggest testing an external source that DOES work, say https://placekitten.com/200/200 (ie test that with valid image sources that it doesn't raise an error, but instead returns a PDF.

Reply:
Agreed, done.